clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -1897,10 +1979,18 @@ def SwitchOp : CIR_Op<"switch",
 
   let assemblyFormat = [{
     custom<SwitchOp>(
-      $regions, $cases, $condition, type($condition)
+      $body, $condition, type($condition)
     )
     attr-dict
   }];
+
+  let extraClassDeclaration = [{
+    // collect cases in the switch.
+    void collectCases(llvm::SmallVector<CaseOp> &cases);
+
+    // check if the switch is in a simple form. If yes, collect the cases to \param cases.
+    bool isSimpleForm(llvm::SmallVector<CaseOp> &cases);
```

Comment:
Might be worth adding a note that this is potentially expensive and should be used with caution (it's not a state we cache, it's computed every time we look at the switch).